### PR TITLE
Optimize payFee 

### LINKED ISSUE #2002: Optimize payFees
``` miner.payFees,11.946809 OK```

### PR DESCRIPTION
**Pull request by peterlimg**
## Fixes
- closes #2002
## Changes
Optimized the `payFees` SC from `8ms` to `1.7ms`

- Do not load all miners and sharders for rewarding, load ids and get random ones first, then load selected nodes.
- Do not load all miners and sharders for view change pool work to delete nodes, save node ids that will be deleted and remove them all at once.
- Change num sharder rewarded each round from 5 to 1 and delegate pool from 1 to 5. Saving sharder nodes and stake pools are slow, so instead of saving 5 nodes and each node reward 1 pool, we can save 1 node and each node reward 5 pools. 

## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/smartcontract/minersc/delete.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+	_, err = state.InsertTrieNode(dKey, &ids)
+	return err
+}
+
+func resetDeleteNodeIDs(state state.StateContextI, key string) error {
+	_, err := state.InsertTrieNode(key, &NodeIDs{})
+	return err
+}
+
+//func resetDeleteSharderIDs(state state.StateContextI) error {
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/minersc/delete.go]: clean up commented code

**Review by dabasov — CHANGES_REQUESTED** [review]
lgtm

**Review by dabasov — APPROVED** [review]
lgtm

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | TRUE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | optimize-pay-fees  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**